pkg/core/runtime/builder.go

```diff
@@ -74,6 +76,8 @@ type Builder struct {
 	metrics        metrics.Metrics
 	erf            events.ListenerFactory
 	apim           api_server.APIManager
+	xdsauth        xds_auth.Authenticator
+	xdsCallbacks   util_xds.Callbacks
```

Comment:
I'm not sure it's quite the refactoring that's needed, even though naming may suggest it. The HDS server uses `XDSAuthenticator` because it uses xDS but it doesn't use `XDSCallbacks` because that's only for the Envoy xDS server.

Reply:
Feels like these 2 are meant to be used together. It's a shame they can't be defined together.

Reply:
Yeah, I'm not quite sure what the right interface/refactoring is here. The callbacks are only needed in case the authenticator needs to take action when a proxy disconnects.

Perhaps next to this new XDS context (the third "xds context" next to `pkg/xds/context.Context` and `pkg/xds/server/v3.Context`) I can create an HDSContext as well.

Reply:
IMO this refactoring entails more than just these two parts of the larger context. Would it be ok to tackle this in a separate PR?

Reply:
Yes, definitely separate PR.